stats/stats-server/src/server.rs

```diff
@@ -134,3 +221,178 @@ pub async fn stats(settings: Settings) -> Result<(), anyhow::Error> {
 
     launcher::launch(&launch_settings, http_router, grpc_router).await
 }
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use tokio::{task::JoinSet, time::error::Elapsed};
+    use wiremock::{
+        matchers::{method, path},
+        Mock, MockServer, ResponseTemplate,
+    };
+
+    use super::*;
+
+    async fn mock_indexing_status(response: ResponseTemplate) -> MockServer {
+        let mock_server = MockServer::start().await;
+        Mock::given(method("GET"))
+            .and(path("/api/v2/main-page/indexing-status"))
+            .respond_with(response)
+            .mount(&mock_server)
+            .await;
+        mock_server
+    }
+
+    async fn test_wait_indexing(
+        wait_config: StartConditionSettings,
+        response: ResponseTemplate,
+    ) -> Result<Result<(), anyhow::Error>, Elapsed> {
+        let server = mock_indexing_status(response).await;
+        tokio::time::timeout(
+            Duration::from_millis(500),
+            wait_for_blockscout_indexing(
+                blockscout_client::Configuration::new(url::Url::from_str(&server.uri()).unwrap()),
+                wait_config,
+            ),
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn wait_for_blockscout_indexing_works_with_200_response() {
+        let wait_config = StartConditionSettings {
+            enabled: true,
+            blocks_ratio_threshold: Some(0.9),
+            internal_transactions_ratio_threshold: Some(0.9),
+            check_period_secs: 0,
+        };
```

Comment:
Nit: maybe extract that in a function, so that you don't need to define the same config in all tests. Otherwise, if something will change in the struct, it may be quite painful to update all definitions

I may also suggest to use `rstest::fixtures` (https://docs.rs/rstest/latest/rstest/attr.fixture.html) in order to avoid defining it in all tests manually